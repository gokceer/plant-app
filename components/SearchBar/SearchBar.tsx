import React from "react";
import { Image, StyleSheet, TextInput, View } from "react-native";

const SearchBar = ({ placeholder }: { placeholder?: string }) => {
  return (
    <View style={styles.searchContainer}>
      <Image
        source={require("../../assets/icons/searchIcon.png")}
        style={styles.searchIcon}
      />
      <TextInput
        placeholder={placeholder || "Search for plants"}
        placeholderTextColor="#AFAFAF"
        style={styles.searchInput}
      />
    </View>
  );
};

export default SearchBar;

const styles = StyleSheet.create({
  searchContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginTop: 14,
    backgroundColor: "#FFFFFFE0",
    borderRadius: 12,
    paddingHorizontal: 12,
    height: 45,
  },
  searchIcon: {
    marginRight: 8,
    width: 20,
    height: 20,
  },
  searchInput: {
    flex: 1,
    fontSize: 16,
  },
});
