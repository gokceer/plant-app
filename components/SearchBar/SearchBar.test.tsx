import { render } from "@testing-library/react-native";
import React from "react";
import SearchBar from "./SearchBar";

describe("SearchBar", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<SearchBar />);
    expect(toJSON()).toMatchSnapshot();
  });
});
