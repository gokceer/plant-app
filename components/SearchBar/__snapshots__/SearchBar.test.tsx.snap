// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFFE0",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 45,
      "marginTop": 14,
      "paddingHorizontal": 12,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/icons/searchIcon.png",
      }
    }
    style={
      {
        "height": 20,
        "marginRight": 8,
        "width": 20,
      }
    }
  />
  <TextInput
    placeholder="Search for plants"
    placeholderTextColor="#AFAFAF"
    style={
      {
        "flex": 1,
        "fontSize": 16,
      }
    }
  />
</View>
`;
